File: JavaAssignment4/src/main/java/TaxiFare.java
public class TaxiFare {
    public Float getFareForDistance(float distance) {
        int firstFiveKm = 5 * 20;
        int tenKmAfterFirstFive = 10 * 18;
        if(distance<0) return -1f;
        if(distance <= 2) return 50.0f;
        else if(distance <= 5) return distance * 20;
        else if(distance <= 15) return  firstFiveKm + (distance-5) * 18;
        else return firstFiveKm + tenKmAfterFirstFive + (distance - 15) * 15;
    }
}

File: Assignment 4/Assignment4Typscript/TaxiFare.ts
export class TaxiFare{
    
    first_five_km = 5 * 20;
    ten_km_after_first_five = 10 * 18;

    getFare(distance: number): number {    
        if (distance==null||distance<=0) {
            return -1;
        } else if (distance<=2) {
            return 50.0
        }else if (distance<=5) {
            return distance*20
        }else if (distance<=15) {
            return this.first_five_km + (distance - 5) * 18
        }else {
            return this.first_five_km + this.ten_km_after_first_five + (distance - 15) * 15
        }
    }
}